### PR TITLE
A handful of fixes to make the Cheffish matrix pass on all the relevant Chef versions.

### PR DESCRIPTION
**Pull request by randomcamel**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -15,7 +16,8 @@ env:
   - CHEF_VERSION=12.4.1
   - CHEF_VERSION=12.3.0
   - CHEF_VERSION=12.2.1
-  - CHEF_VERSION=12.1.2
+  # - CHEF_VERSION=12.2.0
+  # - CHEF_VERSION=12.1.2    # known broken for now.
```
Comment (tyler-ball) [inline .travis.yml]: Did the `allow_failures` block not work for these?  I would think we still want it testing and showing up as failures (unless its taking a lot of time or something)

**Comment by tyler-ball** [comment]
I have 1 question but :+1: and feel free to merge after answering that

File: .travis.yml (hunk 1 above)
Comment (randomcamel) [inline .travis.yml]: I'll clean that up: I'm thinking we don't need 12.2.0 if we've got 12.2.1, and 12.1.2 is broken but we can run it anyway. I'll also re-add Ruby versions to test against.

File: .travis.yml [hunk 2]
```diff
@@ -3,9 +3,9 @@ language: ruby
 cache: bundler
 
 rvm:
-  # - 2.0.0
-  # - 2.1.1
-  - 2.1.2
+  - 2.0.0
```
Comment (jkeiser) [inline .travis.yml]: Not required for this checkin; but if you just say `2.0` and `2.1`, I think Travis will pick the latest.

File: .travis.yml [hunk 3]
```diff
@@ -15,11 +15,10 @@ env:
   - CHEF_VERSION=12.4.1
   - CHEF_VERSION=12.3.0
   - CHEF_VERSION=12.2.1
-  - CHEF_VERSION=12.1.2
+  - CHEF_VERSION=12.1.2    # known broken due to ffi-yajl versioning issues.
```
Comment (jkeiser) [inline .travis.yml]: Note: I am still testing, but I think you can avoid the rvm / chef_version cross by doing something like this: https://github.com/chef/ruby-project-template/blob/master/.travis.yml#L20

Basically, use `include:` in the Matrix to include CHEF_VERSION ./ rvm pairs, and don't bother pairing different CHEF_VERSIONs with different RVMs. "Matrix" becomes "test latest released chef against Ruby 2.0 and Ruby 2.1", and "test master and 12.1-12.3 against Ruby 2.1." Eliminates Ruby 2.0 + master/12.1-12.3 combos.

**Comment by jkeiser** [comment]
:+1:

**Pull request merged**